libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
     value = _udp_ota->peek();
     if(value < '0' || value > '9'){
       data[index++] = '\0';
       return atoi(data);
     }
-    data[index++] = _udp_ota->read();
+    data[index] = _udp_ota->read();
   }
   return 0;
 }
 
 String ArduinoOTAClass::readStringUntil(char end){
   String res = "";
-  int value;
+  char value;
   while(true){
     value = _udp_ota->read();
-    if(value < 0 || value == '\0' || value == end){
+    if(value < '0' || value == '\0' || value == end){
```

Comment:
This change needs to be undon. The check value < 0 was meant to cover an error return from peek() or read(). Undo.